lib/utils/fetch.ts

```diff
@@ -1,6 +1,9 @@
 // https://eckertalex.dev/blog/typescript-fetch-wrapper
-async function http<T>(path: string, config?: RequestInit): Promise<T> {
-    const request = new Request(path, { headers: { 'Content-Type': 'application/json' }, ...config });
+async function http<T>(path: string, config?: RequestInit, auth?: string): Promise<T> {
+    const request = new Request(path, {
+        headers: { 'Content-Type': 'application/json', Authorization: `Bearer ${auth}` },
+        ...config,
+    });
```

Comment:
Can't you just init the `const idJwt = useUser()?.getSignInUserSession()?.getIdToken().getJwtToken();` variable here and set the Authorization header if it's not null? Then you wouldn't need to get the token on every form and pass it along to this function

Reply:
`import { useUser } from '@/lib/hooks/useAuth';

// https://eckertalex.dev/blog/typescript-fetch-wrapper
async function http<T>(path: string, config?: RequestInit): Promise<T> {
    const idJwt = useUser()?.getSignInUserSession()?.getIdToken().getJwtToken();
    const request = new Request(path, {
        headers: { 'Content-Type': 'application/json', Authorization: `Bearer ${idJwt}` },
        ...config,
    });
    const response = await fetch(request);`

Doesn't seem to work

Reply:
How about 
```typescript
import { Auth } from 'aws-amplify';
...
const authSession = await Auth.currentSession();
const jwt = authSession.getIdToken()?.getJwtToken();
```

Reply:
That works